crates/starknet-devnet/tests/test_estimate_fee.rs

```diff
@@ -569,4 +570,63 @@ mod estimate_fee_tests {
             .iter()
             .for_each(assert_fee_estimation);
     }
+
+    #[tokio::test]
+    async fn unsuccessful_declare_update_gas_successful_declare() {
+        let devnet = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let (signer, account_address) = devnet.get_first_predeployed_account().await;
+        let predeployed_account = Arc::new(SingleOwnerAccount::new(
+            devnet.clone_provider(),
+            signer.clone(),
+            account_address,
+            constants::CHAIN_ID,
+            ExecutionEncoding::New,
+        ));
+        let (contract_class, casm_class_hash) =
+            get_simple_contract_in_sierra_and_compiled_class_hash();
+
+        let max_gas_fee = FieldElement::from(1e14 as u128);
+
+        let unsuccessful_declare_tx = predeployed_account
+            .declare(Arc::new(contract_class.clone()), casm_class_hash)
+            .max_fee(max_gas_fee)
+            .send()
+            .await;
+
+        match unsuccessful_declare_tx {
+            Err(AccountError::Provider(ProviderError::StarknetError(
+                StarknetError::InsufficientMaxFee,
+            ))) => (),
+            other => panic!("Unexpected result: {other:?}"),
+        };
+
+        let wei_price = 9e8 as u128;
+        let strk_price = 7e8 as u128;
+        let gas_update = json!({
+            "gas_price_wei": 9e8 as u128,
+            "data_gas_price_wei": 8e8 as u128,
+            "gas_price_strk": 7e8 as u128,
+            "data_gas_price_strk": 6e8 as u128,
+            "generate_block": false,
```

Comment:
So the user will need to specify all prices if they want to change just one?

Reply:
I can propose option parameters, so users could update one or more gas prices

Reply:
e3744dff1e7b3b4fed9f507f17818d21776a45a2